src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/impl/LagOperationServiceImpl.java

```diff
@@ -0,0 +1,203 @@
+import org.openkilda.persistence.repositories.SwitchPropertiesRepository;
+import org.openkilda.persistence.repositories.SwitchRepository;
+import org.openkilda.persistence.tx.TransactionManager;
+import org.openkilda.wfm.topology.switchmanager.error.InconsistentDataException;
+import org.openkilda.wfm.topology.switchmanager.error.InvalidDataException;
+import org.openkilda.wfm.topology.switchmanager.error.SwitchNotFoundException;
+import org.openkilda.wfm.topology.switchmanager.service.LagOperationService;
+
+import com.google.common.collect.Sets;
+import com.google.common.collect.Sets.SetView;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class LagOperationServiceImpl implements LagOperationService {
+    private final TransactionManager transactionManager;
+    private final SwitchRepository switchRepository;
+    private final SwitchPropertiesRepository switchPropertiesRepository;
+    private final LagLogicalPortRepository lagLogicalPortRepository;
+    private final PhysicalPortRepository physicalPortRepository;
+    private final IslRepository islRepository;
+    private final FlowRepository flowRepository;
+    private final FlowMirrorPathRepository flowMirrorPathRepository;
+    private final int bfdPortOffset;
+    private final int bfdPortMaxNumber;
+    private final int lagPortOffset;
+
+    public LagOperationServiceImpl(RepositoryFactory repositoryFactory, TransactionManager transactionManager,
+                                   int bfdPortOffset, int bfdPortMaxNumber, int lagPortOffset) {
+        this.transactionManager = transactionManager;
+        this.switchRepository = repositoryFactory.createSwitchRepository();
+        this.switchPropertiesRepository = repositoryFactory.createSwitchPropertiesRepository();
+        this.lagLogicalPortRepository = repositoryFactory.createLagLogicalPortRepository();
+        this.physicalPortRepository = repositoryFactory.createPhysicalPortRepository();
+        this.islRepository = repositoryFactory.createIslRepository();
+        this.flowRepository = repositoryFactory.createFlowRepository();
+        this.flowMirrorPathRepository = repositoryFactory.createFlowMirrorPathRepository();
+        this.bfdPortOffset = bfdPortOffset;
+        this.bfdPortMaxNumber = bfdPortMaxNumber;
+        this.lagPortOffset = lagPortOffset;
+    }
+
+    @Override
+    public int createLagPort(SwitchId switchId, List<Integer> physicalPortNumbers) {
+        int lagLogicalPortNumber = LagLogicalPort.generateLogicalPortNumber(physicalPortNumbers, lagPortOffset);
+        LagLogicalPort lagLogicalPort = new LagLogicalPort(switchId, lagLogicalPortNumber);
+        List<PhysicalPort> physicalPorts = physicalPortNumbers.stream()
```

Comment:
It must be done inside `LagLogicalPort` (into constructor or into setter), in other case it can't guarantee data consistency. Or at least `LagLogicalPort` object must validate switchId and logicalPortNumber for passed physical ports.